mod/01_core/format/html_format.rb

```diff
@@ -216,8 +265,8 @@ def unsupported_view view
     end
 
     def final_link href, opts={}
-      text = opts[:text] || href
-      %{<a class="#{opts[:class]}" href="#{href}">#{text}</a>}
+      text = (opts.delete(:text) || href).dup
+      content_tag :a, text, opts.merge(:href=>href)
```

Comment:
My comments were not about encoding.  Encoding is great.  But you changed a bunch of tests from `http://localhost:2020/path?cgi=foo&bar=baz` to `http://localhost:2020/path?cgi=foo+bar=baz`. That's not an encoding change, it's just a screwed up url.  The parameters aren't even parseable.  I gather this is fixed now?

I think I was wrong about build_link needing to handle url_options; in general the assumption is that this is handled before we get there.  (I do like the idea of moving the text into the options hash, but that's not vital).  This definitely makes sense given the different contexts that we want to use this.

The reason I got confused was that link_to_page did handle url_options for :type and :view.   That was a bit of a hack, but I'm concerned that this could mean there's a hidden bug now.